modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMFormInstanceRecordLocalServiceTest.java

```diff
@@ -133,11 +212,52 @@ private void _assertDDMFormInstanceRecord(
 			expectedValue, value.getString(value.getDefaultLocale()));
 	}
 
+	private HttpServletRequest _getHttpServletRequest() throws Exception {
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.setAttribute(
+			JavaConstants.JAKARTA_PORTLET_RESPONSE,
+			new MockLiferayPortletRenderResponse());
+		mockHttpServletRequest.setAttribute(
+			WebKeys.CURRENT_URL, "http://localhost:8080/currentURL");
```

Comment:
@MarinhoFeliphe is it possible to use the interface here? see [brian's feedback](https://github.com/brianchandotcom/liferay-portal/commit/91e643e656bba8a4f6f7065a0b5fb8f495f585a1)